packages/solvers/Dockerfile

```diff
@@ -18,7 +18,7 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
     g++ zlib1g-dev libboost-all-dev flex bison
 
 # Install Z3
-COPY --from=ghcr.io/z3prover/z3:ubuntu-20.04-bare-z3-sha-770c51a /usr/bin/z3 /usr/local/bin/z3
+COPY --from=ghcr.io/z3prover/z3:ubuntu-20.04-bare-z3-20240922 /usr/bin/z3 /usr/local/bin/z3
```

Comment:
z3 --version says 4.13.1

Reply:
hmm, no release tag for 4.13.1: https://github.com/Z3Prover/z3/releases

Reply:
it seems that the z3 docker images aren't synced with release versions. no latest version is available as a package:
https://github.com/z3prover/z3/pkgs/container/z3/versions

i think it would be more reliable to install from the github release files, as we do for yices and cvc5.

Reply:
in [9ad0087](https://github.com/a16z/halmos/pull/404/commits/9ad0087bc6a5446c8b520abcad1461659ff60309)